test/SlowTests/Client/TimeSeries/Query/TimeSeriesJavascriptProjections.cs

```diff
@@ -0,0 +1,2008 @@
+                    Assert.Equal("Oren Ayende", result[0].Name);
+
+                    Assert.Equal(3, result[0].Heartrate.Count);
+
+                    var agg = result[0].Heartrate.Results;
+
+                    Assert.Equal(2, agg.Length);
+
+                    Assert.Equal(79, agg[0].Max[0]);
+                    Assert.Equal(69, agg[0].Avg[0]);
+
+                    Assert.Equal(179, agg[1].Max[0]);
+                    Assert.Equal(179, agg[1].Avg[0]);
+
+                }
+            }
+        }
+
+        [Fact]
+        public void TimeSeriesAggregationInsideJsProjection_UsingLinq_FromLoadedDocument()
+        {
+            using (var store = GetDocumentStore())
+            {
+                var baseline = DateTime.Today;
+
+                using (var session = store.OpenSession())
+                {
+                    var companyId = "companies/1";
+                    session.Store(new Person
+                    {
+                        Name = "Oren",
+                        LastName = "Ayende",
+                        Age = 30,
+                        WorksAt = companyId
+                    });
+                    session.Store(new Company
+                    {
+                        Name = "HR",
+                    }, companyId);
+
+                    var tsf = session.TimeSeriesFor(companyId);
+
+                    tsf.Append("Stock", baseline.AddMinutes(61), "tags/1", new[] { 12.59d });
+                    tsf.Append("Stock", baseline.AddMinutes(62), "tags/1", new[] { 12.79d });
+                    tsf.Append("Stock", baseline.AddMinutes(63), "tags/2", new[] { 12.69d });
+
+                    tsf.Append("Stock", baseline.AddMonths(1).AddMinutes(61), "tags/1", new[] { 13.59d });
+                    tsf.Append("Stock", baseline.AddMonths(1).AddMinutes(62), "tags/2", new[] { 13.79d });
+                    tsf.Append("Stock", baseline.AddMonths(1).AddMinutes(63), "tags/1", new[] { 13.69d });
+
+                    session.SaveChanges();
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var query = from p in session.Query<Person>()
+                                where p.Age > 21
+                                let company = RavenQuery.Load<Company>(p.WorksAt)
+                                select new
+                                {
+                                    Heartrate = RavenQuery.TimeSeries(company, "Stock", baseline, baseline.AddMonths(2))
+                                        .Where(ts => ts.Tag == "tags/1")
+                                        .GroupBy(g => g.Months(1))
+                                        .Select(g => new
+                                        {
+                                            Avg = g.Average(),
+                                            Max = g.Max()
+                                        })
+                                        .ToList(),
```

Comment:
Is the ToList needed? 

I really like what you have done here, the API is beautiful